vars/piperLoadGlobalExtensions.groovy

```diff
@@ -0,0 +1,106 @@
+ * Instead of defining the extensions in the .pipeline folder the extensions are defined in another repository.
+ * You can also place a file called extension_configuration.yml in this repository.
+ * Configuration defined in this file will be treated as default values with a lower precedence then custom defaults defined in the project configuration.
+ * You can also define additional Jenkins libraries these extensions depend on using a yaml file called sharedLibraries.yml:
+ * Example:
+ * - name: my-extension-dependency
+ *   version: git-tag
+ */
+@GenerateDocumentation
+void call(Map parameters = [:]) {
+
+    handlePipelineStepErrors (stepName: STEP_NAME, stepParameters: parameters) {
+        def script = checkScript(this, parameters)
+        // load default & individual configuration
+        Map configuration = ConfigurationHelper.newInstance(this)
+            .loadStepDefaults()
+            .mixinGeneralConfig(script.commonPipelineEnvironment, GENERAL_CONFIG_KEYS)
+            .mixin(parameters, PARAMETER_KEYS)
+            .use()
+
+        echo configuration.toString()
+        if(!configuration.globalExtensionsRepository){
+            return
+        }
+
+        dir(configuration.globalExtensionsDirectory){
+            Map gitParameters = [
+                $class: 'GitSCM',
+                userRemoteConfigs: [[url: configuration.globalExtensionsRepository]]
+            ]
+
+            if(configuration.globalExtensionsRepositoryCredentialsId){
+                gitParameters.userRemoteConfigs[0].credentialsId = configuration.globalExtensionsRepositoryCredentialsId
+            }
+
+            if(configuration.globalExtensionsVersion){
+                gitParameters.branches = [[name: configuration.globalExtensionsVersion]]
+            }
+
+            checkout(gitParameters)
+        }
+
+        String extensionConfigurationFilePath = "${configuration.globalExtensionsDirectory}/extension_configuration.yml"
+        if (fileExists(extensionConfigurationFilePath)) {
+            writeFile file: ".pipeline/extension_configuration.yml", text: readFile(file: extensionConfigurationFilePath)
+            DebugReport.instance.globalExtensionConfigurationFilePath = extensionConfigurationFilePath
+
+            prepareDefaultValues([
+                script: script,
+                customDefaults: parameters.customDefaults,
+                customDefaultsFromFiles: ['extension_configuration.yml'] + parameters.customDefaultsFromFiles
+            ])
+        }
+
+        def globalExtensionsLibraryConfig = "${configuration.globalExtensionsDirectory}/sharedLibraries.yml"
+
+        if(fileExists(globalExtensionsLibraryConfig)){
+            loadLibrariesFromFile(globalExtensionsLibraryConfig)
+        }
+    }
+}
+
+private loadLibrariesFromFile(String filename) {
+    List libs = readYaml file: filename
```

Comment:
The contents of this file will be user-input, so perhaps move this into a try-catch-block and give helpful error output? Also, doesn't readYaml always return a Map?